### PR TITLE
Generate and load sample data

### PR DESCRIPTION
**Pull request by medinam6**
Please review my schema and db-data files so I may merge with main. I think just one review is sufficient, so we can all add a review to our deliverables. Thank you

### CODE REVIEW

File: server/db/db-data.js [hunk 1]
```diff
@@ -0,0 +1,124 @@
+const mathRandom = function getRandomInt(minParam, maxParam) {
+  const min = Math.ceil(minParam);
+  const max = Math.floor(maxParam);
+
+  return Math.floor(Math.random() * (max - min + 1) + min);
+};
+
+const authors = function pickAuthors() {
+  const names = ['Gillian C.', 'Chapman U.', 'Isaiah L.', 'Blair B.', 'Bentle P.', 'Menzie F.', 'Alexander R.', 'Walton S.', 'Rosemary D.', 'Wheeler I.', 'Britney P.', 'Morton M.', 'Damian E.', 'Duncan B.', 'Stan E.', 'Brooks A.', 'Ward O.', 'Benson T.', 'Ferris S.', 'Mitchell E.', 'Ulderico O.', 'Toscani Y.', 'Fiorenza B.', 'Lombardi V.', 'Amando X.'];
+  const index = mathRandom(0, 24);
```
Comment (achenqi) [inline server/db/db-data.js]: Recommend using .length - 1 instead so you can modify your arrays without worrying about modifying the range.

File: server/db/db-data.js [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+const sellers = function generateSellers(sellersParam) {
+  const index = mathRandom(0, 71);
+  const listOfSellers = sellersParam.split(' ');
+
+  return listOfSellers[index] + listOfSellers[index + 1];
+};
+
+const text = function createQuestionOrAnswer(textParam) {
+  const start = mathRandom(0, 300);
+  const end = mathRandom(500, 350);
```
Comment (achenqi) [inline server/db/db-data.js]: Unsure how this one works,

Comment (achenqi) [reply server/db/db-data.js]: Math.random() *(350 - 500 + 1) +500) = Math.random() * (-149) + 500,

Comment (achenqi) [reply server/db/db-data.js]: Math.random() * (500 - 350 + 1) + 350 = Math.random() * (151) + 350, looks to be the same nvm.

File: server/db/db-data.js [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+  let questionOrAnswer = textParam.slice(start, end);
```
Comment (achenqi) [inline server/db/db-data.js:70]: Another way to separate the spaces and periods is using string.split(' ')

File: server/db/index.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  database: 'tptqanda',
+  password: 'cdw@20TW'
+});
+
+connection.connect((err) => {
+  if (err) throw err;
+  console.log('Connected to mysql server!');
+});
+
+function promiseQueries(info) {
```
Comment (achenqi) [inline server/db/index.js:21]: Was this function used?

File: server/db/index.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+  info.map(x => {
+    return 'INSERT INTO questions SET ?'
+  })
+}
+
+async function loadData() {
+  try {
+    const questionQuery = await connection.query('INSERT INTO questions (id, productId, seller, date, author, question, flag) VALUES ?', [questions]);
```
Comment (achenqi) [inline server/db/index.js]: This is cool, didn't know query allowed arrays

File: server/db/db-data.js [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+  let id = 1;
+  let numberOfQs;
+  let seller;
+  for (let i = 1; i < 101; i += 1) {
+    numberOfQs = mathRandom(3, 5);
+    seller = sellers(string);
+    for (let j = 0; j < numberOfQs; j += 1) {
+      const questionArr = [
+        id,
+        i,
```
Comment (achenqi) [inline server/db/db-data.js:99]: It looks like you are inserting one question per product, might need to change this later to have more questions per product. Although I know we're only supposed to have 100 records.

File: server/index.js [hunk 7]
```diff
@@ -1,9 +1,23 @@
...
 
-app.listen(port, function() {
+const corsOptions = {
+  origin: 'http://127.0.0.1:5500',
+  optionsSuccessStatus: 200,
+};
+
+app.use(express.static(path.join(__dirname, 'client/dist')));
+
+app.get('/questions', cors(corsOptions), (req, res) => {
```
Comment (achenqi) [inline server/index.js:15]: Add params so that it can retrieve based on productid

**Review by achenqi — COMMENTED** [review]
Great job Meyvi

File: server/db/db-data.js (hunk 2 above)
Comment (medinam6) [reply server/db/db-data.js]: Thank you!

File: server/db/index.js (hunk 4 above)
Comment (medinam6) [reply server/db/index.js:21]: no, I deleted it

File: server/db/db-data.js (hunk 6 above)
Comment (medinam6) [reply server/db/db-data.js:99]: The first loop iterates through all 100 products and the second loop will iterate through 3-6 questions/answers.

**Pull request closed without merging**